modules/src/core/ics04_channel/events.rs

```diff
@@ -256,7 +256,7 @@ impl From<Attributes> for Vec<Tag> {
         if let Some(channel_id) = a.channel_id {
             let channel_id = Tag {
                 key: CHANNEL_ID_ATTRIBUTE_KEY.parse().unwrap(),
-                value: channel_id.as_str().parse().unwrap(),
```

Comment:
`.to_string().parse()` looks like an unnecessary string conversion roundtrip. But an ABCI tag value is a UTF-8 string super-encoded as base64 for an inexplicable reason. Go figure.

Reply:
Right, it seems unnecessary. But I don't think there's any other public constructor for `Tag::Value`. 🤔 So not sure how to fix this. IIUC, the tag value was originally designed to be opaque bytes and later it was decided that (event) indexing was required and we're now left with this base64 encoded string.

Reply:
Nothing to fix here, maybe just add a comment?

Reply:
Just noticed that we already have this comment above -> https://github.com/informalsystems/ibc-rs/pull/2071/files/91c4c04dd191753eeac90159d2912262dfee17e0#diff-5d54d6bafcc46a42f95443d7c324b7c79563ad1f089618a686a8eb0d54e8beb0L237-L242

```rust
/// # Note
/// The parsing of `Key`s and `Value`s never fails, because the
/// `FromStr` instance of `tendermint::abci::tag::{Key, Value}`
/// is infallible, even if it is not represented in the error type.
/// Once tendermint-rs improves the API of the `Key` and `Value` types,
/// we will be able to remove the `.parse().unwrap()` calls.
```

Reply:
Yes, nothing to fix there, it's just an observation (maybe we should have an etiquette about this :blush:)